Auth0/UserInfo.swift

```diff
@@ -0,0 +1,140 @@
+        self.sub = sub
+
+        self.name = name
+        self.givenName = givenName
+        self.familyName = familyName
+        self.middleName = middleName
+        self.nickname = nickname
+        self.preferredUsername = preferredUsername
+
+        self.profile = profile
+        self.picture = picture
+        self.website = website
+
+        self.email = email
+        self.emailVerified = emailVerified
+
+        self.gender = gender
+        self.birthdate = birthdate
+
+        self.zoneinfo = zoneinfo
+        self.locale = locale
+
+        self.phoneNumber = phoneNumber
+        self.phoneNumberVerified = phoneNumberVerified
+        self.address = address
+
+        self.updatedAt = updatedAt
+
+        self.additional = additional
+    }
+
+    convenience required public init?(json: [String: Any]) {
+        guard let sub = json["sub"] as? String else { return nil }
+
+        let name = json["name"] as? String
+        let givenName = json["given_name"] as? String
+        let familyName = json["family_name"] as? String
+        let middleName = json["middle_name"] as? String
+        let nickname = json["nickname"] as? String
+        let preferredUsername = json["preferred_username"] as? String
+
+        var profile: URL?
+        if let profileURL = json["profile"] as? String { profile = URL(string: profileURL) }
+
+        var picture: URL?
+        if let pictureURL = json["picture"] as? String { picture = URL(string: pictureURL) }
+
+        var website: URL?
+        if let websiteURL = json["website"] as? String { website = URL(string: websiteURL) }
+
+        let email = json["email"] as? String
+        let emailVerified = json["email_verified"] as? Bool
+
+        let gender = json["gender"] as? String
+        let birthdate = json["birthdate"] as? String
+
+        var zoneinfo: TimeZone?
+        if let timeZone = json["zoneinfo"] as? String { zoneinfo = TimeZone(identifier: timeZone) }
+
+        var locale: Locale?
+        if let localeInfo = json["locale"] as? String { locale = Locale(identifier: localeInfo) }
+
+        let phoneNumber = json["phone_number"] as? String
+        let phoneNumberVerified = json["phone_number_verified"] as? Bool
+        let address = json["address"] as? [String: String]
+
+        var updatedAt: Date?
+        if let dateString = json["updated_at"] as? String {
+            updatedAt = date(from: dateString)
+        }
+
+        let keys = Set(["sub", "name", "given_name", "family_name", "middle_name", "nickname", "preferred_username", "profile", "picture", "website", "email", "email_verified", "gender", "birthdate", "zoneinfo", "locale",
```

Comment:
I agree `publicClaims` should be static.
The issue with `filter` here is it will return an array of tuples, so it's not quite as elegant as this one liner.
In Swift 4 this works as a filtered dictionary will return a dictionary. (As it should really)